src/Web5.js

```diff
@@ -0,0 +1,223 @@
+import nacl from 'tweetnacl';
+import * as DwnSDK from '@tbd54566975/dwn-sdk-js';
+
+import { DIDRegistry } from './did/DIDRegistry.js';
+import { Protocols } from './interface/Protocols.js';
+import { Records } from './interface/Records.js';
+import { LocalStorage } from './storage/LocalStorage.js';
+import { AppTransport } from './transport/AppTransport.js';
+import { HTTPTransport } from './transport/HTTPTransport.js';
+import { decodePin, parseJSON, triggerProtocolHandler } from './utils.js';
+
+class Web5 extends EventTarget {
+  #did;
+  #interfaces;
+  #transports;
+
+  #keys = null;
+  #connection = null;
+
+  constructor() {
+    super();
+
+    this.#did = new DIDRegistry();
+
+    this.#interfaces = {
+      protocols: new Protocols(this),
+      records: new Records(this),
+    };
+
+    this.#transports = {
+      app: new AppTransport(),
```

Comment:
Similar question to interfaces but since the transports are stateless and could be static methods, what is the benefit of using classes and having to instantiate instances?

Reply:
was just thinking about this. a `WebSocketTransport` may not be stateless